testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/forms/BrowserFlowTest.java

```diff
@@ -190,10 +214,258 @@ public void userWithTwoAdditionalFactors() throws InterruptedException {
+                                .addAuthenticatorExecution(Requirement.ALTERNATIVE, OTPFormAuthenticatorFactory.PROVIDER_ID)
+                        )
+                )
+                .defineAsBrowserFlow()
+        );
+    }
+
+    // In a form waiting for a username only, provides a username and check if password is requested in the following execution of the flow
+    private boolean needsPassword(String username) {
+        // provides username
+        loginUsernameOnlyPage.open();
+        loginUsernameOnlyPage.login(username);
+
+        return passwordPage.isCurrent();
+    }
+
+    // A conditional block without conditional block should automatically be disabled
+    @Test
+    public void testFlowDisabledWhenConditionalBlockIsMissing() {
+        try {
+            configureBrowserFlowWithConditionalSubFlowHavingConditionalBlock("browser - non missing conditional block", true);
+            Assert.assertTrue(needsPassword("user-with-two-configured-otp"));
+
+            configureBrowserFlowWithConditionalSubFlowHavingConditionalBlock("browser - missing conditional block", false);
+            // Flow is conditional but it is missing a conditional authentication executor
+            // The whole flow is disabled
+            Assert.assertFalse(needsPassword("user-with-two-configured-otp"));
+        } finally {
+            testingClient.server("test").run(setBrowserFlowToRealm());
+        }
+    }
+
+    private void configureBrowserFlowWithConditionalSubFlowHavingConditionalBlock(String newFlowAlias, boolean conditionFlowHasConditionalBlock) {
+        testingClient.server("test").run(session -> FlowUtil.inCurrentRealm(session).copyBrowserFlow(newFlowAlias));
+        testingClient.server("test").run(session -> FlowUtil.inCurrentRealm(session)
+                .selectFlow(newFlowAlias)
+                .inForms(forms -> forms
+                        .clear()
+                        .addAuthenticatorExecution(Requirement.REQUIRED, UsernameFormFactory.PROVIDER_ID)
+                        .addSubFlowExecution(Requirement.CONDITIONAL, subFlow -> {
+                            if (conditionFlowHasConditionalBlock) {
+                                // Add authenticators to this flow: 1 conditional block and a basic authenticator executions
+                                subFlow.addAuthenticatorExecution(Requirement.REQUIRED, ConditionalBlockUserConfiguredAuthenticatorFactory.PROVIDER_ID);
+                            }
+                            // Update the browser forms only with a UsernameForm
+                            subFlow.addAuthenticatorExecution(Requirement.REQUIRED, PasswordFormFactory.PROVIDER_ID);
+                        }))
+                .defineAsBrowserFlow()
+        );
+    }
+
+    // Configure a conditional block in a non-conditional sub-flow
+    // In such case, the whole sub-flow should be disabled
```

Comment:
It should be "// In such case, the conditional block is not evaluated", no? And this is what this test shows